src/main/java/staffconnect/model/person/predicates/PersonHasAvailabilitiesPredicate.java

```diff
@@ -0,0 +1,60 @@
+package staffconnect.model.person.predicates;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import staffconnect.commons.util.ToStringBuilder;
+import staffconnect.model.availability.Availability;
+import staffconnect.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Availabilities} matches any of the
+ * availabilities given.
+ */
+public class PersonHasAvailabilitiesPredicate implements Predicate<Person> {
+    private final Set<Availability> availabilities;
+
+    public PersonHasAvailabilitiesPredicate(Set<Availability> availabilities) {
+        this.availabilities = availabilities;
+    }
```

Comment:
I think a simple way to allow you test for partial abilities would be to transform your input of ```Set<Availability>``` into a ```List<Set<String>>``` Availabilities Split into Keywords. 
Where you take each availability in the set and split the inner string by spaces, so each component of an availability is the keyword in a ``` Set<String>```. 
Then you can test each of the keywords with the corresponding functions like ```stream::anyMatch``` etc. This predicate can then be used in the ```find```  command as well, and thus would not need to add any additional methods to ```ParseUtil``` to parse partial availabilities. Let me know what you think of this suggestion

Reply:
yes, i think this may work fine, thank you for the suggestion!
will try it out in a future PR.
can i ask what you mean by "This predicate can then be used in the find command as well"? as in, you rather the partial value filtering be part of the `find` command or?

Reply:
What I mean is that this ```PersonHasAvailabilitesPredicate```  can then be reused in both here and the ```find``` command